File: src/test/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderTrustAccountRepositoryTest.java
package uk.gov.justice.hmpps.prison.repository.jpa.repository;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.jdbc.test.autoconfigure.AutoConfigureTestDatabase;
import org.springframework.boot.data.jpa.test.autoconfigure.DataJpaTest;
import org.springframework.test.context.ActiveProfiles;
import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderTrustAccount;
import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderTrustAccount.Pk;

import static org.assertj.core.api.Assertions.assertThat;
import static org.springframework.boot.jdbc.test.autoconfigure.AutoConfigureTestDatabase.Replace.NONE;

@DataJpaTest
@ActiveProfiles("test")
@AutoConfigureTestDatabase(replace = NONE)
public class OffenderTrustAccountRepositoryTest {
    @Autowired
    private OffenderTrustAccountRepository repository;

    @Test
    void testOffenderTrustAccountMapping() {
        final var optionalOffenderTrustAccount = repository.findById(new Pk("LEI", -1001L));

        assertThat(optionalOffenderTrustAccount).get().extracting(OffenderTrustAccount::getAccountClosedFlag).isEqualTo("N");
    }
}

File: src/test/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderTrustAccountRepositoryTest.kt
package uk.gov.justice.hmpps.prison.repository.jpa.repository

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.data.jpa.test.autoconfigure.DataJpaTest
import org.springframework.boot.jdbc.test.autoconfigure.AutoConfigureTestDatabase
import org.springframework.test.context.ActiveProfiles
import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderTrustAccountId

@DataJpaTest
@ActiveProfiles("test")
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
class OffenderTrustAccountRepositoryTest {
  @Autowired
  private lateinit var repository: OffenderTrustAccountRepository

  @Test
  fun testOffenderTrustAccountMapping() {
    val optionalOffenderTrustAccount = repository.findById(OffenderTrustAccountId("LEI", -1001L)).get()

    assertThat(optionalOffenderTrustAccount.accountClosed).isFalse
  }
}
